### ISSUE TITLE
[addon fold/foldcode] HTML folding shows multiple arrows when opening tag spans multiple lines.

### ISSUE THREAD

**Issue by takamori** [codemirror/CodeMirror#5179]
It looks weird and is confusing. It would be more acceptable if it behaved similar to other languages, even tho ideally it would show the fold arrow for the first line of the open tag and show the collapsed content double-ended arrow starting from the end of the open tag.

1) Open http://codemirror.net/demo/folding.html 
2) For the HTML example, add a line break on the div before the id attribute. Watch closely to see what happened to the fold arrows.
3) Look at what happens as one plays with the fold arrows in that section.

The same doesn't occur w/ the other languages, from what I could tell.

**Comment by marijnh** [comment]
Attached patch should fix this.

**Comment by takamori** [comment]
I just noticed that this commit makes the result worse, actually. Currently the result is that no fold appears at all when there's a line break; I compared the behavior if I revert the change using DevTools to modify the code.

**Comment by marijnh** [comment]
I see. Does attached patch help?

**Comment by takamori** [comment]
Uh... now it has the originally-filed behavior where two arrows show up, but that's better than the behavior after the first patch.

To be clear, the originally-filed issue is that the following content:
```
<div title="foo"
         id="bar">
   <span>some content</span>
</div>
```
gets collapsed into the following appearance:
```
35▾   <div title="foo"
36▸         id="bar">↔</div>
```
rather than the following (note that there is no extra arrow at line 35):
```
35    <div title="foo"
36▸         id="bar">↔</div>
```

**Comment by marijnh** [comment]
I don't see two arrows showing up anymore, when I paste the snippet you show. Are you sure you're testing the current code?

**Comment by takamori** [comment]
I've tried using DevTools to "Empty cache and hard reload" the folding demo page, and it repros. Not when you immediately paste the snippet- but when you click the arrow in the gutter to collapse the item, the two arrows appear. And clicking on either of those two arrows results in expanding only the div element.

**Comment by tyrw** [comment]
Unfortunately this is not fixed as of 5.47.0, which incorporates the above commit by AlexanderPrendota.  Suggest to re-open this issue.

You can reproduce it with the official code folding demo: https://codemirror.net/demo/folding.html

Make any HTML tag multi-line, and the error persists on folding:

![Screen Shot 2019-06-03 at 10 43 11 AM](https://user-images.githubusercontent.com/819876/58822505-8912d680-85ec-11e9-9760-dba2b84c1473.png)
